src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -460,7 +460,7 @@ private static DetailAST getPreviousNodeWithParentOfTypeAst(DetailAST ast, Detai
         // checked whether ident or lastTypeNode has preceding position
         // determining if it is java style or C style
 
-        if (ident == null || isTypeCast || ident.getLineNo() > ast.getLineNo()) {
```

Comment:
We do not need this null check anymore, it was from this the inconsistent `TYPE` node that we are eliminating:

Code like:

```java

        if (info != null)
            return (TypeVariable<Class<String>>[])info.getTypeParameters();
        else
            return (TypeVariable<Class<String>>[])new TypeVariable<?>[0];
```

If we place a print statement inside this condition:

```java

        if (ident == null || isTypeCast || ident.getLineNo() > ast.getLineNo()) {
            if (ident == null && !isTypeCast) {
                System.out.println(lastTypeNode);
            }

```

We were only catching the case where the type argument child was NOT an `IDENT`:

```

>[161x45]
>[163x45]

```



From master:

```
...
    |       |   |   |       |   |--IDENT -> TypeVariable [161:20]
    |       |   |   |       |   `--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [161:32]
    |       |   |   |       |       |--GENERIC_START -> < [161:32]
    |       |   |   |       |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [161:33]
    |       |   |   |       |       |   `--TYPE -> TYPE [161:33]
    |       |   |   |       |       |       |--IDENT -> Class [161:33]
    |       |   |   |       |       |       |--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [161:38]
    |       |   |   |       |       |       |   |--GENERIC_START -> < [161:38]
    |       |   |   |       |       |       |   |--TYPE_ARGUMENT -> TYPE_ARGUMENT [161:39]
    |       |   |   |       |       |       |   |   `--TYPE -> TYPE [161:39]
    |       |   |   |       |       |       |   |       `--IDENT -> String [161:39]
    |       |   |   |       |       |       |   `--GENERIC_END -> > [161:45]
    |       |   |   |       |       |       `--ARRAY_DECLARATOR -> [ [161:47]
    |       |   |   |       |       |           `--RBRACK -> ] [161:48]
    |       |   |   |       |       `--GENERIC_END -> > [161:46]
...

```

From 10121:

```

...
    |       |   |   |       |   |--IDENT -> TypeVariable [161:20]
    |       |   |   |       |   `--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [161:32]
    |       |   |   |       |       |--GENERIC_START -> < [161:32]
    |       |   |   |       |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [161:33]
    |       |   |   |       |       |   |--IDENT -> Class [161:33]
    |       |   |   |       |       |   |--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [161:38]
    |       |   |   |       |       |   |   |--GENERIC_START -> < [161:38]
    |       |   |   |       |       |   |   |--TYPE_ARGUMENT -> TYPE_ARGUMENT [161:39]
    |       |   |   |       |       |   |   |   `--IDENT -> String [161:39]
    |       |   |   |       |       |   |   `--GENERIC_END -> > [161:45]
    |       |   |   |       |       |   `--ARRAY_DECLARATOR -> [ [161:47]
    |       |   |   |       |       |       `--RBRACK -> ] [161:48]
    |       |   |   |       |       `--GENERIC_END -> > [161:46]
 ...

```

Now we are able to get the `IDENT` from the `TYPE_ARGUMENT` ast.